src/Features/VisualBasic/Portable/PopulateSwitch/VisualBasicPopulateSwitchCodeFixProvider.vb

```diff
@@ -0,0 +1,137 @@
+﻿Imports System.Composition
+Imports System.Runtime.InteropServices
+Imports Microsoft.CodeAnalysis.CodeFixes
+Imports Microsoft.CodeAnalysis.CSharp.CodeFixes.PopulateSwitch
+Imports Microsoft.CodeAnalysis.Formatting
+Imports Microsoft.CodeAnalysis.Simplification
+Imports Microsoft.CodeAnalysis.Text
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.CodeFixes.PopulateSwitch
+    <ExportCodeFixProvider(LanguageNames.VisualBasic, Name:=PredefinedCodeFixProviderNames.PopulateSwitch), [Shared]>
+    <ExtensionOrder(After:=PredefinedCodeFixProviderNames.AddOverloads)>
+    Partial Friend Class VisualBasicPopulateSwitchCodeFixProvider
+        Inherits AbstractPopulateSwitchCodeFixProvider
+
+        Protected Overrides Function GetSwitchExpression(node As SyntaxNode) As SyntaxNode
+
+            Dim selectBlock = DirectCast(node, SelectBlockSyntax)
+            Return selectBlock.SelectStatement.Expression
+        End Function
+
+        Protected Overrides Function GetMissingLabels(node As SyntaxNode, model As SemanticModel, enumType As INamedTypeSymbol, <Out> ByRef containsDefaultLabel As Boolean) As List(Of String)
+
+            Dim caseLabels = GetCaseLabels(DirectCast(node, SelectBlockSyntax), containsDefaultLabel)
+            Dim symbols As New List(Of ISymbol)
+
+            For Each label In caseLabels
+
+                ' these are the labels like `MyEnum.EnumMember`
+                Dim memberAccessExpression = TryCast(label, MemberAccessExpressionSyntax)
```

Comment:
Same comment as for C# case.  If you simplify the C# code it seems like you could move this method entirely up into the base class.